pkg/sideinputs/utils/utils.go

```diff
@@ -0,0 +1,69 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+)
+
+// CheckFileExists checks if a file with the given fileName exists in the file system.
+func CheckFileExists(fileName string) bool {
+	_, err := os.Stat(fileName)
+	// check if err is "file not exists"
+	return !os.IsNotExist(err)
+}
+
+// UpdateSideInputFile writes the given side input value to a new file
+// and updates the side input store path to point to this new file.
+func UpdateSideInputFile(ctx context.Context, fileSymLink string, value []byte) error {
+	log := logging.FromContext(ctx)
+	// Generate a new file name using timestamp
+	timestamp := time.Now().Unix()
+	newFileName := fmt.Sprintf("%s_%d", fileSymLink, timestamp)
+
+	// Write the side input value to the new file
+	err := os.WriteFile(newFileName, value, 0666)
+	if err != nil {
+		return fmt.Errorf("failed to write Side Input file %s : %w", newFileName, err)
+	}
+
+	// Get the old file path
+	oldFilePath, _ := os.Readlink(fileSymLink)
```

Comment:
Do we need to check the existence of `fileSymLink`?

Reply:
You mean if this returns an error or not?
If there is an error we should get back a empty string and then at the time of deleting the file we check whether it exists or not before removing.